### PR TITLE
Order by primary keys ascending if simple query sql without order by clause for MySQL

### LINKED ISSUE #9109: MySQL InnoDB back-end ShardingSphere-Proxy query result of sql without order by clause is not sorted by primary key ascending
### Which version of ShardingSphere did you use?
5.0.0-RC1-SNAPSHOT

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy with MySQL 5.7.29

### Expected behavior
Sort order is the same as MySQL

### Actual behavior
Result not sorted

### Reason analyze (If you can)
Query sql is not rewritten as `ORDER BY`

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
config-sharding.yaml snippet:
```
rules:
- !SHARDING
 tables:
   t_order:
     actualDataNodes: ds_${0..1}.t_order_${0..1}
```
table schema:
```
CREATE TABLE `t_order_0 / 1` (
  `order_id` int(11) NOT NULL,
  `user_id` int(11) NOT NULL,
  `status` varchar(45) DEFAULT NULL,
  PRIMARY KEY (`order_id`)
) ENGINE=InnoDB;
```
example query result:
```
mysql> SELECT * FROM t_order WHERE order_id>=1 AND order_id<=10;
+----------+---------+---------------------------+
| order_id | user_id | status                    |
+----------+---------+---------------------------+
|        2 |      94 | inited 2 at 16:47:44.588  |
|        4 |      34 | inited 4 at 16:47:44.607  |
|        6 |      66 | inited 6 at 16:47:44.628  |
|       10 |      80 | inited 10 at 16:47:44.668 |
|        1 |      20 | inited 1 at 16:47:44.571  |
|        5 |      36 | inited 5 at 16:47:44.619  |
|        7 |      50 | inited 7 at 16:47:44.636  |
|        8 |     101 | inited 8 at 16:47:44.645  |
|        3 |      73 | inited 3 at 16:47:44.598  |
|        9 |      23 | inited 9 at 16:47:44.656  |
+----------+---------+---------------------------+
```

### PR DESCRIPTION
**Pull request by sandynz**
Fixes #9109.

Changes proposed in this pull request:
- Order by primary keys ascending if simple query sql without order by clause for MySQL (just support simple query sql: single table, no join tables, no subquery, no aggregation, table has primary key)

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/pom.xml [hunk 1]
```diff
@@ -43,5 +43,12 @@
             <artifactId>shardingsphere-infra-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        
+        <dependency>
+            <groupId>org.apache.shardingsphere</groupId>
+            <artifactId>shardingsphere-sql-parser-mysql</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/pom.xml:52]: The infra module can not understand concrete module such as shardingsphere-sql-parser-mysql

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by sandynz** [comment]
Blocked on integration test MySQL-Proxy, aggregation function with `DISTINCT` test failed on `Placeholder` mode, e.g.
```
Error case: assertExecuteQuery[proxy: db -> MySQL -> Placeholder -> SELECT SUM(DISTINCT order_id) s FROM t_order WHERE order_id < 1100]

java.lang.AssertionError: Size of actual result set is different with size of expected dat set rows.
Expected: is <1>
     but: was <0>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.apache.shardingsphere.test.integration.engine.it.dql.BaseDQLIT.assertRows(BaseDQLIT.java:102)
	at org.apache.shardingsphere.test.integration.engine.it.dql.BaseDQLIT.assertResultSet(BaseDQLIT.java:71)
	at org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT.assertExecuteQueryForPreparedStatement(GeneralDQLIT.java:81)
	at org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT.assertExecuteQuery(GeneralDQLIT.java:62)
```

**Comment by terrymanu** [comment]
> Blocked on integration test MySQL-Proxy, aggregation function with `DISTINCT` test failed on `Placeholder` mode, e.g.
> 
> ```
> Error case: assertExecuteQuery[proxy: db -> MySQL -> Placeholder -> SELECT SUM(DISTINCT order_id) s FROM t_order WHERE order_id < 1100]
> 
> java.lang.AssertionError: Size of actual result set is different with size of expected dat set rows.
> Expected: is <1>
>      but: was <0>
> 
> 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
> 	at org.junit.Assert.assertThat(Assert.java:956)
> 	at org.apache.shardingsphere.test.integration.engine.it.dql.BaseDQLIT.assertRows(BaseDQLIT.java:102)
> 	at org.apache.shardingsphere.test.integration.engine.it.dql.BaseDQLIT.assertResultSet(BaseDQLIT.java:71)
> 	at org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT.assertExecuteQueryForPreparedStatement(GeneralDQLIT.java:81)
> 	at org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT.assertExecuteQuery(GeneralDQLIT.java:62)
> ```

We can ignore that case because it is known case, I will fix it soon

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java [hunk 2]
```diff
@@ -78,4 +96,42 @@ private OrderByContext createOrderByContextForDistinctRowWithoutGroupBy(final Se
         }
         return null;
     }
+    
+    private OrderByContext createOrderByContextForMySQLSelectWithoutOrderBy(final ShardingSphereSchema schema, final SelectStatement selectStatement, final GroupByContext groupByContext) {
+        if (groupByContext.getItems().size() > 0) {
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java:101]: `groupByContext.getItems().size() > 0` can be replace to `!groupByContext.getItems().isEmpty()`

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java [hunk 3]
```diff
@@ -78,4 +96,42 @@ private OrderByContext createOrderByContextForDistinctRowWithoutGroupBy(final Se
         }
         return null;
     }
+    
+    private OrderByContext createOrderByContextForMySQLSelectWithoutOrderBy(final ShardingSphereSchema schema, final SelectStatement selectStatement, final GroupByContext groupByContext) {
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java:100]: It is better to refactor return null to return `Optional`

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java [hunk 4]
```diff
@@ -78,4 +96,42 @@ private OrderByContext createOrderByContextForDistinctRowWithoutGroupBy(final Se
...
+        
+        for (ProjectionSegment projectionSegment : selectStatement.getProjections().getProjections()) {
+            if (projectionSegment instanceof AggregationProjectionSegment) {
+                return null;
+            }
+        }
+        
+        int index = 0;
+        List<OrderByItem> orderByItems = new LinkedList<>();
+        for (String primaryKeyColumn : tableMetaData.getPrimaryKeyColumns()) {
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/orderby/engine/OrderByContextEngine.java:126]: `primaryKeyColumn` should rename to `each`

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**